keps/sig-api-machinery/5073-declarative-validation-with-validation-gen/README.md

```diff
@@ -0,0 +1,1458 @@
+      - [Phase3: Finalization and GA (Core Team and community)](#phase3-finalization-and-ga-core-team-and-community)
+    - [Tagging and Validating Against Internal vs Versioned Types](#tagging-and-validating-against-internal-vs-versioned-types)
+    - [Handling Zero Values in Declarative Validation](#handling-zero-values-in-declarative-validation)
+      - [Difficulties with <code>+required</code> and <code>+default</code>](#difficulties-with-required-and-default)
+      - [Proposed Solutions](#proposed-solutions)
+      - [Addressing the Problem with Valid Zero Values Using the Linter](#addressing-the-problem-with-valid-zero-values-using-the-linter)
+    - [Ratcheting](#ratcheting)
+    - [Test Plan](#test-plan)
+        - [Prerequisite testing updates](#prerequisite-testing-updates)
+        - [Unit tests](#unit-tests)
+        - [Integration tests](#integration-tests)
+        - [e2e tests](#e2e-tests)
+    - [Graduation Criteria](#graduation-criteria)
+      - [Beta](#beta)
+        - [DeclarativeValidation](#declarativevalidation)
+        - [DeclarativeValidation](#declarativevalidation-1)
+    - [Upgrade / Downgrade Strategy](#upgrade--downgrade-strategy)
+    - [Version Skew Strategy](#version-skew-strategy)
+  - [Production Readiness Review Questionnaire](#production-readiness-review-questionnaire)
+    - [Feature Enablement and Rollback](#feature-enablement-and-rollback)
+    - [Rollout, Upgrade and Rollback Planning](#rollout-upgrade-and-rollback-planning)
+    - [Monitoring Requirements](#monitoring-requirements)
+    - [Dependencies](#dependencies)
+    - [Scalability](#scalability)
+    - [Troubleshooting](#troubleshooting)
+  - [Implementation History](#implementation-history)
+  - [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+    - [Use CEL and OpenAPI libraries directly for K8s Native Types (<a href="https://github.com/kubernetes/enhancements/tree/master/keps/sig-api-machinery/4153-declarative-validation">KEP-4153</a>)](#use-cel-and-openapi-libraries-directly-for-k8s-native-types-kep-4153)
+  - [Infrastructure Needed (Optional)](#infrastructure-needed-optional)
+<!-- /toc -->
+
+
+# KEP-5073: Declarative Validation of Kubernetes Native Types With validation-gen
+
+
+## Summary
+
+
+    With this enhancement, Kubernetes Developers will declare validation rules using Interface Definition Language (IDL) tags in the `types.go` files that define the Kubernetes native API types. For example:
+
+
+```go
+// staging/src/k8s.io/api/core/v1/types.go
+
+// ReplicationControllerSpec is the specification of a replication controller.
+type ReplicationControllerSpec struct {
+        // +k8s:optional
+        // +k8s:minimum=0
+        Replicas *int32 `json:"replicas,omitempty" protobuf:"varint,1,opt,name=replicas"`
+
+        // +k8s:optional
+        // +k8s:minimum=0
+        MinReadySeconds int32 `json:"minReadySeconds,omitempty" protobuf:"varint,4,opt,name=minReadySeconds"`
+...
+}
+```
```

Comment:
The quoting here and in several places below isn't quite correct, see [rendered document](https://github.com/kubernetes/enhancements/blob/c1bce5ae0d5fc114f0a23c1e0e7f59e24a86f03e/keps/sig-api-machinery/5073-declarative-validation-with-validation-gen/README.md).